rust/Cargo.toml

```diff
@@ -45,9 +44,9 @@ rusoto_dynamodb = { version = "0.48", default-features = false, optional = true
 rusoto_glue = { version = "0.48", default-features = false, optional = true }
 
 # Datafusion
-datafusion = { version = "12", optional = true }
-datafusion-expr = { version = "12", optional = true }
-datafusion-common = { version = "12", optional = true }
+datafusion = { git = "https://github.com/spaceandtimelabs/arrow-datafusion.git", rev = "cb9241982556745939175b0597194ad819bab613", optional = true }
```

Comment:
This will point back at `apache/arrow-datafusion` once that PR (#3867) is merged. Then probably `v14.0.0` once it is released.